app/services/core_data_connector/search/base.rb

```diff
@@ -322,7 +322,11 @@ def to_search_json(skip_relationships = true)
 
             # Extract the value for the attribute
             if attr[:block].present?
-              value = instance_eval(&attr[:block])
+              if name == :geometry
```

Comment:
Any reason we only want to do this for the `:geometry` attribute? We could make the options available to all attributes and allow the attributes to decide which make sense to use.